packages/react-dom-bindings/src/server/fizz-instruction-set/ReactDOMFizzInstructionSet.js

```diff
@@ -131,15 +132,22 @@ export function completeBoundaryWithStyles(
     }
   }
 
-  Promise.all(dependencies).then(
+  const allInsertions = Promise.all(dependencies).then(
     completeBoundaryImpl.bind(null, suspenseBoundaryID, contentID, ''),
     completeBoundaryImpl.bind(
       null,
       suspenseBoundaryID,
       contentID,
-      'Resource failed to load',
+      'Stylesheet failed to load',
     ),
   );
+
+  if (isContainer) {
+    const containerNode = document.getElementById(suspenseBoundaryID);
+    if (containerNode) {
+      containerNode['_r'] = allInsertions;
+    }
+  }
```

Comment:
This is how we "block" hydration until the boundary can switch the content in. There are still an unhandled edge cases though

For fizz runtime the script loads async so it's possible the bootstrap scripts execute first. This means a hydrateRoot call might start to compare HTML before the actual content is swapped in. I don't really have a good idea atm how to defend against this. It may require coordination with the container itself (an attribute that is removed when hydration is unblocked)

cc @sebmarkbage